packages/fury-adapter-oas3-parser/lib/parser/oas/parseOperationObject.js

```diff
@@ -11,25 +13,46 @@ const parseObject = require('../parseObject');
 const parseString = require('../parseString');
 const parseResponsesObject = require('./parseResponsesObject');
 const parseParameterObjects = require('./parseParameterObjects');
+const parseRequestBodyObject = require('./parseRequestBodyObject');
+const parseReference = require('../parseReference');
+
+const parseRequestBodyObjectOrRef = parseReference('requestBodies', parseRequestBodyObject);
 
 const name = 'Operation Object';
 const requiredKeys = ['responses'];
 const unsupportedKeys = [
-  'tags', 'externalDocs', 'requestBody', 'callbacks', 'deprecated', 'security',
+  'tags', 'externalDocs', 'callbacks', 'deprecated', 'security',
 ];
 const isUnsupportedKey = R.anyPass(R.map(hasKey, unsupportedKeys));
 
+const isRequestBody = R.allPass([isMember, hasKey('requestBody')]);
+
 function createTransactions(namespace, member, operation) {
+  const requests = R.map(getValue, R.filter(isRequestBody, operation.content));
   const responses = operation.get('responses');
 
-  return responses.map((response) => {
-    const request = new namespace.elements.HttpRequest();
-    const method = member.key.clone();
-    method.content = method.content.toUpperCase();
-    request.method = method;
+  if (requests.length === 0) {
+    requests.push(new namespace.elements.HttpRequest());
+  }
+
+  const transactions = [];
 
-    return new namespace.elements.HttpTransaction([request, response]);
+  requests.forEach((request) => {
+    responses.forEach((response) => {
```

Comment:
I went down an approach to refactor this to be a bit more functional, however I think the clarity of using nested forEach makes the code more easier to follow.